packages/@ember/-internals/metal/lib/property_set.ts

```diff
@@ -70,6 +75,18 @@ export function set(obj: object, keyName: string, value: any, tolerant?: boolean
   let descriptor = descriptorForProperty(obj, keyName);
 
   if (descriptor !== undefined) {
+    if (DEBUG) {
+      let instanceDesc = lookupDescriptor(obj, keyName);
+      let name = instanceDesc?.set?.toString();
+
+      assert(
+        `Attempted to set \`${toString(
+          obj
+        )}.${keyName}\` using Ember.set(), but the property was a computed or tracked property that was shadowed by another property declaration. This can happen if you defined a tracked or computed property on a parent class, and then redefined it on a subclass.`,
+        name && (name.indexOf('CPSETTER_FUNCTION') !== -1 || name.indexOf('TRACKED_SETTER') !== -1)
```

Comment:
Instead of indexing the name can we keep the weakset system (for debug only)?

Reply:
we can, it's just more of a pain to maintain

Reply:
also FWIW what we do for `CPGETTER_FUNCTION`: https://github.com/emberjs/ember.js/blob/bugfix/assert-when-setting-shadowed-property/packages/@ember/-internals/metal/lib/decorator.ts#L103